File: transita-mais-backend-priv/transita-commons/src/main/java/br/com/texsistemas/transitamais/commons/api/dto/ponto/PontoOnibusDTO.java
package br.com.texsistemas.transitamais.commons.api.dto.ponto;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public record PontoOnibusDTO(
        UUID id,
        String titulo,
        String descricao,
        Float avaliacao,
        String informacao,
        EnderecoDTO endereco,
        BigDecimal latitude,
        BigDecimal longitude,
        List<HorarioDTO> horarios,
        String tipoPonto,
        String linkImagem
) {
}

File: mobile/app/src/main/java/br/com/texsistemas/transita/data/web/dto/PontoOnibusDTO.kt
package br.com.texsistemas.transita.data.web.dto

import kotlinx.serialization.Serializable

@Serializable
data class PontoOnibusDTO(
    val id: String,
    val titulo: String,
    val descricao: String,
    val avaliacao: Double,
    val informacao: String,
    val endereco: EnderecoDTO,
    val latitude: Double,
    val longitude: Double,
    val horarios: List<HorarioDTO>,
    val tipoPonto: String,
    val linkImagem: String?
)

@Serializable
data class EnderecoDTO(
    val id: String,
    val rua: String,
    val numero: String,
    val bairro: String,
    val cidade: String,
    val uf: String,
    val cep: String
)

@Serializable
data class HorarioDTO(
    val id: String,
    val horario: String,
    val tipoPonto: String
)